File: backend/app/schemas/adp.py
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.enums import ADPFormat


class ADPSourceRead(BaseModel):
    source: str
    adp: Decimal

    model_config = ConfigDict(from_attributes=True)


class PlayerADPRead(BaseModel):
    id: UUID
    player_id: UUID
    source: str
    format: ADPFormat
    season: int
    adp: Decimal
    position_rank: int | None

    model_config = ConfigDict(from_attributes=True)


class BatchADPRequest(BaseModel):
    player_ids: list[UUID]
    season: int
    format: ADPFormat | None = None


class ADPSyncResponse(BaseModel):
    synced: int
    skipped: int
    errored: int

File: frontend/src/api/adp.ts
import { get, post } from "./client";

export interface ADPSyncResult {
  synced: number;
  skipped: number;
  errored: number;
}

export function syncADP(season?: number): Promise<ADPSyncResult> {
  const query = season ? `?season=${season}` : "";
  return post<ADPSyncResult>(`/adp/sync${query}`);
}

export interface PlayerADPHistory {
  id: string;
  player_id: string;
  source: string;
  format: string;
  season: number;
  adp: string;
  position_rank: number | null;
}

export function getRosterADP(
  playerIds: string[],
  season: number,
  format?: string,
): Promise<Record<string, string | null>> {
  return post<Record<string, string | null>>("/adp/batch", {
    player_ids: playerIds,
    season,
    format: format || undefined,
  });
}

export function getPlayerADPHistory(
  playerId: string,
  format?: string,
): Promise<PlayerADPHistory[]> {
  const query = format ? `?format=${format}` : "";
  return get<PlayerADPHistory[]>(`/adp/players/${playerId}/history${query}`);
}
